### DISCUSSION TITLE
is there any documentation about the relationship between site url and redirect urls?

### DISCUSSION THREAD

**Discussion by devhandler** [supabase/supabase#15135] [Questions]
there is some in the api doc and redirect wild card. but after extensive search, seems there is no clear doc regarding which is for which use cases. for example:
1. do redirect urls take precedence over site url if they are specified in the signup api json and also configured in the supabase dash?
2. does the site url work more like a catch-all/default thing if no redirect urls are specified in the dash and in api json option?
3. for signup email confirmation, how can we identify it's from the email confirmation link and show some msg like "thanks for confirming your email" when users land on the redirect or site url?

**Comment by GaryAustin1** [comment]
Not sure about docs.  But 
1. Yes
2. Yes
3. You can redirect to a different landing page if you are doing signup with email confirm.  Otherwise would need to look at the session data for times of confirmation versus last sign in time possibly.  I’ve believe I’ve seen discussions here and/or Discord on doing this.

**Reply by devhandler** [reply]
thanks for the help. after i set emailRedirectTo in the api like this: options: {
      emailRedirectTo: 'https://example.com/welcome'
    }
and added exactly the same (also tried globstar **) to the dash redirect url, it doesn't work. when clicking the confirmation link, it still goes to the localhost root specified by the site url in supabase dash...

i also searched various stack overflow posts. some said we need to add a slash at the end of the url, but this doesn't apply to my case because the url i specified in emailRedirectTo is more specific like http://localhost:5173/welcome, which has a slash anyway

**Reply by GaryAustin1** [reply]
Does the redirect_to in the email URL link correct?
If the redirectTo/emailRedirectTo matches exactly a link the the dashboard auth/URL Configuration settings then it should replace the siteURL in the redirect link in the email.

**Reply by devhandler** [reply]
no, the link in the email is not correct. it still goes to sth like http://localhost:5173/#access_token......

while in the auth.user.raw_user_meta_data json field, i can clearly see {emailRedirectTo: http://localhost:5173/auth?s=welcome}, and that url matches exactly the redirect urrl in the url config dash (i copy/pasted it and triple checked)

**Comment by GaryAustin1** [comment]
I think you have the redirect option inside the data option if it is in metadata. It goes on its own.  Sent from my iPhoneOn Jun 18, 2023, at 11:52 AM, devhandler ***@***.***> wrote:﻿
no, the link in the email is not correct. it still goes to sth like http://localhost:5173/#access_token......
while in the auth.user.raw_user_meta_data json field, i can clearly see {emailRedirectTo: http://localhost:5173/auth?s=welcome}, and that url matches exactly the redirect urrl in the url config dash (i copy/pasted it and triple checked)

**Reply by devhandler** [reply]
you are right! i was looking at the doc and thought all the things should go to data field. obviously it should be the top level key of option json.

thank you so much!
